ginivision/src/main/java/net/gini/android/vision/internal/camera/photo/PhotoEdit.java

```diff
@@ -1,88 +1,70 @@
 package net.gini.android.vision.internal.camera.photo;
 
-import android.graphics.Bitmap;
-import android.graphics.BitmapFactory;
 import android.os.AsyncTask;
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 
-import java.io.ByteArrayOutputStream;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * @exclude
  */
 public class PhotoEdit {
 
-    private static final int DEF_QUALITY = 100;
-
     private final Photo mPhoto;
-    private int mRotationDegrees;
-    private int mQuality = DEF_QUALITY;
+    private final List<PhotoModifier> mPhotoModifiers;
 
-    public PhotoEdit(@NonNull Photo photo) {
+    PhotoEdit(@NonNull Photo photo) {
         mPhoto = photo;
-        resetToDefaults();
+        mPhotoModifiers = new ArrayList<>();
```

Comment:
As mentioned above modifications were abstracted away. For each modification a new instance of a specific modifier is created and added to this list. When the modification is applied the list is iterated and the modifications are applied in order.

I kept it really simple.